File: resources/java/Aoc.java
package $package;

import nl.sbmf21.aoc.common.AocBase;

import java.util.HashMap;

public class Aoc extends AocBase {

    public Aoc() {
        super("$year", new HashMap<>());
    }

    public static void main(String[] args) {
        var aoc = new Aoc();

        aoc.exec(args);
    }
}

File: 2022/kotlin/src/main/kotlin/nl/sbmf21/aoc22/Aoc.kt
package nl.sbmf21.aoc22

import nl.sbmf21.aoc.common.AocBase
import nl.sbmf21.aoc.common.buildSimulation
import nl.sbmf21.aoc22.days.simulations.CRT
import nl.sbmf21.aoc22.days.simulations.Crates
import nl.sbmf21.aoc22.days.simulations.LavaDroplet
import nl.sbmf21.aoc22.days.simulations.Snake

class Aoc : AocBase(
    "2022", mapOf(
        "crates" to buildSimulation(::Crates),
        "snake" to buildSimulation(::Snake),
        "crt" to buildSimulation(::CRT),
        "lava" to buildSimulation(::LavaDroplet),
    )
)

fun main(args: Array<String>) = Aoc().exec(args)